File: backend/app/domain/entities/article.py
"""Доменная сущность статьи."""
from datetime import datetime
from typing import Optional, List
from uuid import UUID
from app.domain.entities.base import BaseEntity


class Article(BaseEntity):
    """Статья из arXiv."""
    
    def __init__(
        self,
        arxiv_id: str,
        title: str,
        authors: List[str],
        abstract: str,
        published_date: Optional[datetime] = None,
        categories: Optional[List[str]] = None,
        pdf_url: Optional[str] = None,
        tex_url: Optional[str] = None,
        local_pdf_path: Optional[str] = None,
        local_tex_path: Optional[str] = None,
        parsed_content: Optional[str] = None,
        id: Optional[UUID] = None,
        created_at: Optional[datetime] = None
    ):
        super().__init__(id, created_at)
        self.arxiv_id = arxiv_id
        self.title = title
        self.authors = authors or []
        self.abstract = abstract
        self.published_date = published_date
        self.categories = categories or []
        self.pdf_url = pdf_url
        self.tex_url = tex_url
        self.local_pdf_path = local_pdf_path
        self.local_tex_path = local_tex_path
        self.parsed_content = parsed_content

File: frontend/src/shared/types/article.ts
export type ArticleSummary = {
  id: string
  arxivId: string
  title: string
  authors: string[]
  published: string
  abstract: string
  tags: string[]
}

export type Evaluation = {
  id: string
  articleId: string
  category: string
  relevance: string
  novelty: number
  rigor: number
  impact: number
  overall: number
  pros: string[]
  cons: string[]
  reasoning: string
}

export type ReviewSection = {
  title: string
  body: string
}

export type Review = {
  id: string
  articleId: string
  summary: string
  methods: string
  results: string
  criticism: string
  application: string
  verdict: string
  fullText: string
  sections: ReviewSection[]
}

export type ArticleDetail = ArticleSummary & {
  publishedDate?: string
  pdfUrl?: string
  texUrl?: string
  localPdfPath?: string
  localTexPath?: string
  parsedContent?: string
}
